dags/main_summary.py

```diff
@@ -49,24 +49,30 @@
         }),
     dag=dag)
 
-main_summary = EMRSparkOperator(
+main_summary = MozDatabricksSubmitRunOperator(
     task_id="main_summary",
     job_name="Main Summary View",
     execution_timeout=timedelta(hours=9),
     email=["telemetry-alerts@mozilla.com", "frank@mozilla.com", "main_summary_dataset@moz-svc-ops.pagerduty.com"],
-    instance_count=40,
+    instance_count=5,
+    max_instance_count=40,
+    enable_autoscale=True,
+    instance_type="c4.4xlarge",
+    spot_bid_price_percent=50,
+    ebs_volume_count=1,
+    ebs_volume_size=250,
     env=tbv_envvar("com.mozilla.telemetry.views.MainSummaryView",
         options={
             "from": "{{ ds_nodash }}",
             "to": "{{ ds_nodash }}",
             "schema-report-location": "s3://{{ task.__class__.private_output_bucket }}/schema/main_summary/submission_date_s3={{ ds_nodash }}",
-            "bucket": "{{ task.__class__.private_output_bucket }}"
+            "bucket": "{{ task.__class__.private_output_bucket }}",
+            "read-mode": "aligned",
+            "input-partition-multiplier": "400"
```

Comment:
Can you explain what behavior you expect out of these changes to read-mode and input-partition-multiplier? Are we expecting small dataset behavior due to the high input-partition-multiplier?

Reply:
It's actually the opposite - we expect a behavior change when we switch to autoscaling because the cluster size is initially small, which means that the parallelism is low. Forcing the system to use an exact amount of parallelism will cause scale-up of up to our max number of nodes (but possibly less, hence allowing autoscaling at all).